fee_calculator/apps/calculator/management/commands/generatefees.py

```diff
@@ -407,21 +412,50 @@ def generate_agfs10_fees(agfs_scheme, basic_fees_path, misc_fees_path):
                 fee_type = FeeType.objects.get(code=fee['BIST_BILL_SUB_TYPE'])
 
             for scenario_id in scenario_ids:
-                price = Price.objects.create(
-                    scheme=agfs_scheme,
-                    scenario=Scenario.objects.get(pk=scenario_id),
-                    fee_type=fee_type,
-                    advocate_type=AdvocateType.objects.get(pk=fee['PSTY_PERSON_TYPE']),
-                    offence_class=None,
-                    unit=Unit.objects.get(pk=fee['UNIT']),
-                    fee_per_unit=fee_per_unit,
-                    fixed_fee=fixed_fee,
-                    limit_from=limit_from,
-                    limit_to=limit_to,
-                )
-                if fee['DEFENDANT_UPLIFT_ALLOWED'] == 'Y':
-                    price.modifiers.add(defendant_modifier)
-                if fee['CASE_UPLIFT_ALLOWED'] == 'Y':
-                    price.modifiers.add(case_modifier)
+                if fee['BIST_BILL_SUB_TYPE'] == 'AGFS_CONFERENCE':
+                    for length_limit in conferences_trial_length_limits:
+                        price = Price.objects.create(
+                            scheme=agfs_scheme,
+                            scenario=Scenario.objects.get(pk=scenario_id),
+                            fee_type=fee_type,
+                            advocate_type=AdvocateType.objects.get(pk=fee['PSTY_PERSON_TYPE']),
+                            offence_class=None,
+                            unit=Unit.objects.get(pk=fee['UNIT']),
+                            fee_per_unit=fee_per_unit,
+                            fixed_fee=fixed_fee,
+                            limit_from=length_limit[0],
+                            limit_to=length_limit[1],
+                        )
+                        price.modifiers.add(length_limit[2])
+
+                        if fee['DEFENDANT_UPLIFT_ALLOWED'] == 'Y':
+                            price.modifiers.add(defendant_modifier)
+                        if fee['CASE_UPLIFT_ALLOWED'] == 'Y':
+                            price.modifiers.add(case_modifier)
+
+                        price.save()
+                else:
+                    if fee['BIST_BILL_SUB_TYPE'] in ['AGFS_COMMITTAL', 'AGFS_CONTEMPT']:
+                        # correct erroneous 'HALFDAY' unit for these fees
```

Comment:
corrected now?

Reply:
It was corrected in the actual prices in a previous PR (https://github.com/ministryofjustice/laa-fee-calculator/pull/50), but I thought I should add it to the generation code as well.